AzureFunctions.AngularClient/src/app/site/deployment-center/deployment-center.component.ts

```
@@ -0,0 +1,109 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { BroadcastService } from 'app/shared/services/broadcast.service';
+import { BroadcastEvent } from 'app/shared/models/broadcast-event';
+import { OnDestroy } from '@angular/core/src/metadata/lifecycle_hooks';
+import { TreeViewInfo, SiteData } from 'app/tree-view/models/tree-view-info';
+
+@Component({
+    selector: 'app-deployment-center',
+    templateUrl: './deployment-center.component.html',
+    styleUrls: ['./deployment-center.component.scss']
+})
+export class DeploymentCenterComponent implements OnInit, OnDestroy {
+    public resourceIdStream: Subject<string>;
+    private _resourceId: string;
+    public viewInfoStream = new Subject<TreeViewInfo<SiteData>>();
+    public viewInfo: TreeViewInfo<SiteData>;
+    @Input()
+    set viewInfoInput(viewInfo: TreeViewInfo<SiteData>) {
+        this.viewInfo = viewInfo;
+        this.viewInfoStream.next(viewInfo);
+    }
+
+    get resourceId() {
+        return this._resourceId;
+    }
+    public hasWritePermissions = true;
+
+    private _ngUnsubscribe = new Subject();
+    private _siteConfigObject: ArmObj<SiteConfig>;
+    private _busyManager: BusyStateScopeManager;
+
+    public showFTPDashboard = false;
+    public showWebDeployDashboard = false;
+
+    constructor(
+        private _aiService: AiService,
+        private _authZService: AuthzService,
+        private _cacheService: CacheService,
+        broadcastService: BroadcastService
+    ) {
+        this._busyManager = new BusyStateScopeManager(broadcastService, 'site-tabs');
+
+        this.viewInfoStream
+            .takeUntil(this._ngUnsubscribe)
+            .switchMap(viewInfo => {
+                this._busyManager.setBusy();
+                this._resourceId = viewInfo.resourceId;
+                this._siteConfigObject = null;
+                return Observable.zip(
+                    this._cacheService.getArm(`${this._resourceId}/config/web`),
```

>this._cacheService.getArm(`${this._resourceId}/config/web`), [](start = 20, length = 60)

We should start handling errors similarly to how Ahmed recommended where you never let it bubble up to the top subscription.  Otherwise an exception will completely break your stream and your UI will become unusable.

You can either put a lot of this logic into a service like we discussed, or you can create a private function which handles the exception.  I did something similar in the EmbeddedFunctionEditorComponent._getScriptContent()

---

As mentioned, this is something I'll address, but will be a separate PR

---
In reply to: [161821447](https://github.com/Azure/azure-functions-ux/pull/2205#discussion_r161821447) [](ancestors = 161821447)